src/utils/apiRequests/api.ts

```diff
@@ -1,4 +1,4 @@
-import { TENANT_ID } from '../constants/environment';
```

Comment:
I would suggest passing tenantID to the API handlers (i.e. `getRequest` etc) only if it is set in the QueryParamContext (which indicates it was passed in the URL when the app was first opened).

Over here, the API handlers could use the following logic for the `tenant-key` -

`process.env.TENANT_ID || paramTenantID || DEFAULT_TENANT_ID`

As this would be repeated logic, you could keep a helper function in this file to take care of this.